src/types/vendor/mongoose.d.ts

```diff
@@ -0,0 +1,21 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+/**
+ * Additional type declarations for mongoose to fit our use case,
+ * to accommodate the non-standard but compatible use of types
+ * in schema registration
+ */
+declare module 'mongoose' {
+  namespace Schema {
+    namespace Types {
+      class DocumentArrayWithLooseDiscriminator extends DocumentArray {
+        discriminator<T extends Document>(
+          name: string,
+          schema: Schema<T>,
+          tag?: string,
+        ): any
```

Comment:
does this need to be `any` or can we use `unknown` to be more strict?

Reply:
We can use `unknown`, though I would advise that we line up with mongoose (which uses `any`) as much as we can

Reply:
does using `any` give us any advantage at all though? it just means the return type isn't type-checked, right? it doesn't seem like there's any harm in being more strict than mongoose's types

Reply:
I suppose so - though I would generally prefer to minimise any deviation from the original definition. Let me put in the change now.

Reply:
yeah I agree with that principle generally, but in this case it seems mongoose is using `any` just to make their types easier to work with, but I'd rather do without their generosity because it makes our code less type-safe haha